### PR TITLE
[Model] Add Qwen3-Omni DSpark support

### PR DESCRIPTION
**Pull request by Zhou248**
## Purpose

Add framework support for using a `Qwen3OmniDSparkModel` draft architecture with a Qwen3-Omni thinker target.

This change:
- registers the dedicated Qwen3-Omni DSpark architecture while reusing the shared `Qwen3DSparkForCausalLM` runtime implementation;
- exposes the target thinker's post-DeepStack auxiliary hidden states required by DSpark;
- preserves the dedicated architecture and DSpark sampling fields when converting msModelSpec configuration;
- validates the target/draft hidden size, attention geometry, target-layer selection, vocabulary contract, and DSpark algorithm settings at startup;
- makes embedding and lm-head sharing safe when the draft input/output vocabulary differs from the target vocabulary;
- relies on the draft-logits cache stride fix from #53017 instead of maintaining a model-specific resizing workaround;
- adds focused registry, configuration, model-contract, and configuration-bridge tests.

A repository search for `Qwen3 Omni DSpark` and `Qwen3OmniDSpark` found no existing PR implementing this support, so this does not duplicate known work.

This is a framework integration PR. Qwen3-Omni DSpark checkpoints are not yet available, so it does not claim model quality or acceptance-rate results.

AI assistance disclosure: this PR was prepared with OpenAI Codex assistance. The submitter is responsible for reviewing every changed line and completing the required NPU/model evaluation.

## Test Plan

```bash
.venv/bin/ruff check \
  tests/model_executor/test_qwen3_omni.py \
  tests/models/registry.py tests/test_config.py \
  tests/transformers_utils/test_speculators_dspark_config.py \
  vllm/config/speculative.py \
  vllm/model_executor/models/qwen3_dflash.py \
  vllm/model_executor/models/qwen3_dspark.py \
  vllm/model_executor/models/qwen3_omni_moe_thinker.py \
  vllm/model_executor/models/registry.py \
  vllm/transformers_utils/configs/speculators/algos.py \
  vllm/v1/worker/gpu/spec_decode/dspark/utils.py

.venv/bin/ruff format --check <the same 11 files>

.venv/bin/python -m pytest -q \
  tests/transformers_utils/test_speculators_dspark_config.py \
  tests/test_config.py \
  -k 'qwen3_omni_dspark or dspark_updater'

.venv/bin/python -m pytest -q \
  tests/model_executor/test_qwen3_omni.py \
  -k 'incomplete_vocab_weights'
```

Outstanding NPU/model validation:
- run the Qwen3-Omni hidden-state module tests in an environment with all multimodal dependencies;
- load a compatible Qwen3-Omni DSpark checkpoint;
- run NPU end-to-end eager and ACLGraph speculative-decoding tests in the dependent vLLM-Ascend PR;
- report acceptance length, output correctness, memory, and throughput.

## Test Result

- Ruff check: passed.
- Ruff format check: passed; 11 files already formatted.
- Direct registry smoke test: passed; `Qwen3OmniDSparkModel` resolves to `Qwen3DSparkForCausalLM`.
- Configuration and msModelSpec bridge tests: 21 passed, 201 deselected.
- Vocabulary/weight-contract tests: 3 passed, 3 deselected.
- `git diff --check`: passed.
- `tests/models/test_registry.py` collection is blocked locally by the newly required `model_hosting_container_standards` package; the equivalent direct registry import check passed.
- Model evaluation: not run because no Qwen3-Omni DSpark checkpoint is available.

### CODE REVIEW

File: vllm/model_executor/models/registry.py [hunk 1]
```diff
@@ -637,6 +637,10 @@
     "DFlashMuseGlimmerAssistantModel": ("qwen3_dflash", "DFlashQwen3ForCausalLM"),
     "DSparkDraftModel": ("vllm.models.deepseek_v4", "DSparkDeepseekV4ForCausalLM"),
     "Qwen3DSparkModel": ("qwen3_dspark", "Qwen3DSparkForCausalLM"),
+    "Qwen3OmniDSparkModel": (
+        "qwen3_omni_dspark",
+        "Qwen3OmniDSparkForCausalLM",
+    ),
```
Comment (TheEpicDolphin) [inline vllm/model_executor/models/registry.py]: I think you can just map this to `Qwen3DSparkForCausalLM` instead. `Qwen3OmniDSparkForCausalLM` is just an empty wrapper over `Qwen3DSparkForCausalLM`

File: vllm/v1/worker/gpu/spec_decode/dspark/speculator.py [hunk 2]
```diff
@@ -43,6 +43,20 @@ class DSparkSpeculator(DFlashSpeculator):
...
+        if self.vocab_size != target_vocab_size:
+            self.vocab_size = target_vocab_size
+            if self.draft_logits is not None:
+                self.draft_logits = torch.zeros(
+                    self.max_num_reqs,
+                    self.num_speculative_steps,
+                    self.vocab_size,
+                    dtype=torch.float32,
+                    device=device,
+                )
```
Comment (TheEpicDolphin) [inline vllm/v1/worker/gpu/spec_decode/dspark/speculator.py]: I think this `self.draft_logits` override is no longer needed because of this recently merged PR: https://github.com/vllm-project/vllm/pull/53017. `gumbel_sample` now uses the correct stride from the draft logits cache tensor rather than the vocab size of the logits handed to it, which can be smaller.

**Review by TheEpicDolphin — CHANGES_REQUESTED** [review]
Thanks for the new feature! Left some feedback

File: vllm/v1/worker/gpu/spec_decode/dspark/speculator.py (hunk 2 above)
Comment (Zhou248) [reply vllm/v1/worker/gpu/spec_decode/dspark/speculator.py]: Thanks for the suggestions!! I’ve updated the PR in commit 3e7f8ae5f, could you please take another look?

File: vllm/model_executor/models/registry.py (hunk 1 above)
Comment (Zhou248) [reply vllm/model_executor/models/registry.py]: Thanks for the suggestions!! I’ve updated the PR in commit https://github.com/vllm-project/vllm/commit/3e7f8ae5ffea778770e254774520c25232fa70ca, could you please take another look?

File: vllm/v1/worker/gpu/spec_decode/dspark/speculator.py (hunk 2 above)
Comment (TheEpicDolphin) [reply vllm/v1/worker/gpu/spec_decode/dspark/speculator.py]: @Zhou248 thanks, will take a nother look soon

File: vllm/v1/worker/gpu/spec_decode/dspark/utils.py [hunk 3]
```diff
@@ -62,20 +62,31 @@ def load_dspark_model(target_model: nn.Module, vllm_config: VllmConfig) -> nn.Mo
...
     target_embed = getattr(target_inner, "embed_tokens", None)
     draft_embed = getattr(draft_inner, "embed_tokens", None)
-    if target_embed is not None and _should_share(
-        draft_model, "has_own_embed_tokens", draft_embed, target_embed
+    if (
+        target_embed is not None
+        and draft_model_config.get_vocab_size() == target_vocab_size
+        and _should_share(
+            draft_model, "has_own_embed_tokens", draft_embed, target_embed
+        )
```
Comment (TheEpicDolphin) [inline vllm/v1/worker/gpu/spec_decode/dspark/utils.py:98]: nit: `draft_model_config.get_vocab_size() == target_vocab_size` is perhaps a bit too strict. I think you can do `draft_model_config.get_vocab_size() <= target_vocab_size`. You might have to change `uses_expanded_input_vocab` above to: `uses_expanded_input_vocab = self.config.vocab_size > self.target_vocab_size`.

**Review by TheEpicDolphin — APPROVED** [review]
Thanks for addressing the feedback!

File: vllm/v1/worker/gpu/spec_decode/dspark/utils.py (hunk 3 above)
Comment (Zhou248) [reply vllm/v1/worker/gpu/spec_decode/dspark/utils.py:98]: Thanks for the suggestion! I updated the DSpark embedding sharing condition to allow draft_vocab_size <= target_vocab_size, and changed uses_expanded_input_vocab to only apply when the draft input vocabulary is larger than the target vocabulary.

**Review by ywang96 — APPROVED** [review]
